app/services/wallets/balance/decrease_ongoing_service.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Wallets
+  module Balance
+    class DecreaseOngoingService < BaseService
+      def initialize(wallet:, credits_amount:)
+        super(nil)
+
+        @wallet = wallet
+        @credits_amount = credits_amount
+      end
+
+      def call
+        wallet.update!(
+          ongoing_usage_balance_cents: amount_cents,
+          credits_ongoing_usage_balance: credits_amount,
+          ongoing_balance_cents:,
+          credits_ongoing_balance:,
+        )
+
+        handle_threshold_top_up
+
+        result.wallet = wallet
+        result
+      end
+
+      private
+
+      attr_reader :wallet, :credits_amount
+
+      def handle_threshold_top_up
+        threshold_rule = wallet.recurring_transaction_rules.where(rule_type: :threshold).first
+
+        return if threshold_rule.nil? || wallet.credits_ongoing_usage_balance > threshold_rule.threshold_credits
+
+        WalletTransactions::CreateJob.set(wait: 2.seconds).perform_later(
+          organization_id: wallet.organization.id,
+          wallet_id: wallet.id,
+          paid_credits: threshold_rule.paid_credits.to_s,
+          granted_credits: threshold_rule.granted_credits.to_s,
+          source: :threshold,
+        )
+      end
+
+      def currency
+        @currency ||= wallet.ongoing_balance.currency
+      end
+
+      def amount_cents
+        @amount_cents ||= wallet.rate_amount * credits_amount * currency.subunit_to_unit
+      end
+
+      def pending_transactions
+        @pending_transactions ||= wallet.wallet_transactions.pending
+      end
+
+      def ongoing_balance_cents
+        [
+          0,
+          (pending_transactions.sum(:amount) * currency.subunit_to_unit) - amount_cents + wallet.balance_cents
```

Comment:
Style/TrailingCommaInArrayLiteral: Put a comma after the last item of a multiline array. (https://rubystyle.guide#no-trailing-array-commas)